src/utils/game_utils.ts

```diff
@@ -14,6 +14,9 @@ export function getDidDodge(
   if (attackingMonster.hasAbility(Ability.TRUE_STRIKE)) {
     return false;
   }
+  if (attackingMonster.hasAbility(Ability.SNARE) && attackTarget.hasAbility(Ability.FLYING)) {
```

Comment:
`attackTarget.hasAbility(Ability.FLYING) && attackTarget.hasDebuff(Ability.SNARE)`

When a monster is snared it gets the snare debuff and all monsters will be able to hit it regardless of whether or not the attacking monster has snare. The snare debuff is added before this dodge check

Reply:
It makes sense that the attackTarget monster would have the debuff but they need to actually be hit before that could happen. This check is simply to ensure that the initial hit happens so snare is actually applied and all subsequent attacks from monsters with snare should also hit it 100% of the time.

Some of the logic later with the snare check I believe still applies. Attacking monsters without snare aren't guaranteed to hit a monster with a snare debuff. They can just attack with the same chance to hit as if they didn't have flying. We can get rid of this bit since it can't happen anymore: `!attackingMonster.hasAbility(Ability.SNARE) &&`

Hopefully that all makes sense. TLDR; I think the logic I have on line 17 is still accurate but we can get rid of the redundant snare check on line 33.

Reply:
Got it, I mistakenly read the ability as "all flying monsters that are snared would always hit"